element-hq/element-x-android
# Fix the logic of the room list banner state

## Content

- Make sure `needsSlidingSyncMigration` also generates a state for `derivedStateOf`.
- Fix the logic in the `derivedStateOf` block: it fixes an issue that hid the migration banner, gets rid of the `syncState` and can now be extended with more cases that fallthrough to the next ones.

## Motivation and context

Fix the migration banner not being displayed in some cases that @pixlwave reported. To be honest I have no idea why this issue appeared now, since it was tested several times before and it worked as expected. Maybe an update of the compose compiler broke it?

## Tests

- Log into a HS with both SS and SSS support, with sliding sync disabled in the app (I used `riot.grin.hu`).
- Check the upgrade banner appears.
- It can be dismissed.
- When restarting the app, the banner is displayed again.

## Tested devices

- OS version(s): 14

## Checklist
